spring-restdocs/src/main/java/org/springframework/restdocs/hypermedia/FieldExtractors.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2014-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.hypermedia;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Static factory methods providing a selection of {@link FieldExtractor field extractors}
+ * for use when documentating a hypermedia-based API.
+ *
+ * @author Andreas Evers
+ */
+public abstract class FieldExtractors {
+
+	private FieldExtractors() {
+
+	}
+
+	/**
+	 * Returns a {@code FieldExtractor} capable of extracting fields in Hypermedia
+	 * Application Language (HAL) format where the fields are all properties except
+	 * {@code _links} and {@code _embedded}.
```

Comment:
I could be convinced that `_links` doesn't need to be documented as it could be documented in the same for every resource (a pointer to the links documentation for the resource). However, `_embedded` will vary and, therefore, I suspect that people will want to document it.